src/System.Runtime.Extensions/tests/System/IO/PathTests.Windows.cs

```diff
@@ -0,0 +1,358 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public partial class PathTests_Windows : PathTestsBase
+    {
+        public void GetDirectoryName_DevicePath()
+        {
+            if (PathFeatures.IsUsingLegacyPathNormalization())
+            {
+                Assert.Equal(@"\\?\C:", Path.GetDirectoryName(@"\\?\C:\foo"));
+            }
+            else
+            {
+                Assert.Equal(@"\\?\C:\", Path.GetDirectoryName(@"\\?\C:\foo"));
+            }
+        }
+
+        [Theory, MemberData(nameof(TestData_GetDirectoryName_Windows))]
+        public void GetDirectoryName(string path, string expected)
+        {
+            Assert.Equal(expected, Path.GetDirectoryName(path));
+        }
+
+        [Theory,
+            InlineData("B:", ""),
+            InlineData("A:.", ".")]
+        public static void GetFileName_Volume(string path, string expected)
+        {
+            // With a valid drive letter followed by a colon, we have a root, but only on Windows.
+            Assert.Equal(expected, Path.GetFileName(path));
+        }
+
+        [ActiveIssue(27269, TargetFrameworkMonikers.Netcoreapp)]
```

Comment:
did we get this fix yet?

Reply:
No, unfortunately. I'm not sure why we don't have a new CoreLib, I've run out of mental bandwidth to figure it out. Once we do, we can just remove these few and everything should pass. Well, with https://github.com/dotnet/coreclr/pull/16554 anyway. :)